### PR TITLE
Migrate JDBC driver tests to JUnit

### PR DESCRIPTION
**Pull request by martint**
## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

File: client/trino-jdbc/src/test/java/io/trino/jdbc/TestJdbcConnection.java [hunk 1]
```diff
@@ -451,8 +450,15 @@ public void testSessionUser()
      * @see TestJdbcStatement#testConcurrentCancellationOnStatementClose()
      */
     // TODO https://github.com/trinodb/trino/issues/10096 - enable test once concurrent jdbc statements are supported
-    @Test(timeOut = 60_000, dataProviderClass = DataProviders.class, dataProvider = "trueFalse", enabled = false)
-    public void testConcurrentCancellationOnConnectionClose(boolean autoCommit)
+    @Test(timeOut = 60_000, enabled = false)
```
Comment (dain) [inline client/trino-jdbc/src/test/java/io/trino/jdbc/TestJdbcConnection.java]: Maybe just remove this.  It has been disabled for 2 years

File: client/trino-jdbc/src/test/java/io/trino/jdbc/JdbcDriverIT.java [hunk 2]
```diff
@@ -33,10 +32,10 @@ public class JdbcDriverIT
             "META-INF/MANIFEST.MF",
             "META-INF/services/java.sql.Driver");
 
-    @Parameters("jdbc-jar")
     @Test
-    public void testDependenciesRelocated(String file)
+    public void testDependenciesRelocated()
```
Comment (dain) [inline client/trino-jdbc/src/test/java/io/trino/jdbc/JdbcDriverIT.java:36]: maybe rename this class to `TestJdbcDriverPackaging`

File: client/trino-jdbc/src/test/java/io/trino/jdbc/TestJdbcVendorCompatibility.java [hunk 3]
```diff
@@ -241,69 +212,73 @@ public void testTimestampWithTimeZone()
...
-                    // with calendar
-                    assertEquals(rs.getTimestamp(column, getCalendar()).getTime(), reference.getTimestamp(column, getCalendar()).getTime()); // point in time
-                    assertEquals(rs.getTimestamp(column, getCalendar()), reference.getTimestamp(column, getCalendar()));
-                    assertEquals(rs.getTimestamp(column, getCalendar()), timestampForPointInTime);
-                });
+        try (Connection connection = createConnection();
+                Statement statement = connection.createStatement();
+                ConnectionSetup connectionSetup = new ConnectionSetup(referenceDrivers)) {
+            checkRepresentation(
+                    connection, statement, "TIMESTAMP '1970-01-01 00:00:00.000 +00:00'", // Trino
```
Comment (dain) [inline client/trino-jdbc/src/test/java/io/trino/jdbc/TestJdbcVendorCompatibility.java:219]: I'd either put one argument per line or format like this:
```
checkRepresentation(connection, statement,
        "TIMESTAMP...
```
but feel free to ignore

File: client/trino-jdbc/src/test/java/io/trino/jdbc/TestJdbcWarnings.java [hunk 4]
```diff
@@ -95,82 +96,72 @@ public void setupServer()
         }
     }
 
-    @AfterClass(alwaysRun = true)
+    @AfterAll
     public void tearDownServer()
             throws Exception
     {
         server.close();
         server = null;
```
Comment (dain) [inline client/trino-jdbc/src/test/java/io/trino/jdbc/TestJdbcWarnings.java:104]: not need to null the server with Junit 5

**Review by dain — APPROVED** [review]

File: client/trino-jdbc/src/test/java/io/trino/jdbc/JdbcDriverIT.java (hunk 2 above)
Comment (electrum) [reply client/trino-jdbc/src/test/java/io/trino/jdbc/JdbcDriverIT.java:36]: This is a Maven Failsafe integration test and thus needs to end with `IT`

**Comment by electrum** [comment]
Did you verify that `JdbcDriverIT` runs correctly with Failsafe? I don't know if we configured correctly it for JUnit 5

File: client/trino-jdbc/src/test/java/io/trino/jdbc/TestJdbcVendorCompatibility.java (hunk 3 above)
Comment (martint) [reply client/trino-jdbc/src/test/java/io/trino/jdbc/TestJdbcVendorCompatibility.java:219]: I didn't modify this -- I just added the try block around it

**Comment by martint** [comment]
> Did you verify that JdbcDriverIT runs correctly with Failsafe? I don't know if we configured correctly it for JUnit 5

How do I do that? It passed the CI...

File: client/trino-jdbc/src/test/java/io/trino/jdbc/TestJdbcWarnings.java (hunk 4 above)
Comment (martint) [reply client/trino-jdbc/src/test/java/io/trino/jdbc/TestJdbcWarnings.java:104]: I'll leave all those for further cleanup later. We need to do a sweep through the codebase to update the BeforeAll/AfterAll calls anyway.

**Review by wendigo — APPROVED** [review]

**Pull request merged**